src/client/viewSandbox.js

```diff
@@ -350,15 +353,25 @@ const init = () => {
         } else {
           validateOutput.innerHTML = 'Invalid';
         }
+
+        validateButton.disabled = false;
+        validateButton.textContent = 'Validate';
       })
       .catch(reportIframeCommsError);
   };
 
   const reportSettings = () => {
+    getSettingsButton.disabled = true;
+    copySettingsToInitButton.disabled = true;
+    getSettingsButton.textContent = 'Waiting for the extension settings...';
+
     extensionView
       .getSettings()
       .then(settings => {
         getSettingsEditor.setValue(JSON.stringify(settings, null, 2));
+        getSettingsButton.disabled = false;
+        copySettingsToInitButton.disabled = false;
+        getSettingsButton.textContent = 'Get Settings';
```

Comment:
You should probably run this inside a `finally` so that if there's an error it gets reset.